salalib/shapemap.cpp

```diff
@@ -2534,7 +2546,8 @@ bool ShapeMap::setCurSelDirect(const pvecint& selset, bool add)
    for (size_t i = 0; i < selset.size(); i++) {
       int index = selset[i];  // relies on aligned indices for attributes and shapes
       if (index != -1) {
-         if (m_selection_set.add(index) != -1) {
+         if (std::find(m_selection_set.begin(), m_selection_set.end(), index) == m_selection_set.end()) {

```

Comment:
Maybe some of these should be std::set anyway? If adding is via search, and there's no random access, vectors are not efficient...